packages/framework/client-logger/app-insights-logger/package.json

```diff
@@ -75,6 +75,7 @@
 		"build:compile": "fluid-build . --task compile",
 		"build:docs": "fluid-build . --task api",
 		"build:esnext": "tsc-multi --config ../../../../common/build/build-common/tsc-multi.esm.json",
+		"build:test": "tsc --project ./tsconfig.json",
```

Comment:
What exactly is the error if we remove this script? The package doesn't have a separate tests-only build like others do (with a separate tsconfig file inside src/test, that gets used for this script). So I'm thinking of either removing this script if we can skip that specific policy for this package, or changing it to something like `"build:test": "echo 'No separate tests-only build'"` (which might also require some policy exclusion).

Reply:
It was like:
~~~
client-release-group-root: error during command 'flub check policy' (exit code 1)
client-release-group-root: WARNING:
client-release-group-root: file failed the "fluid-build-tasks-tsc" policy: packages/framework/client-logger/app-insights-logger/package.json
client-release-group-root: 'tsc' task is missing the following dependency:
client-release-group-root:      typetests:gen
client-release-group-root: Statistics: 7844 processed, 107 excluded, 7951 total
client-release-group-root: Performance for "handle":
client-release-group-root:      fluid-case: 3074ms
client-release-group-root:      dockerfile-copyright-file-header: 1ms
client-release-group-root:      js-ts-copyright-file-header: 128ms
client-release-group-root:      npm-package-metadata-and-sorting: 547ms
client-release-group-root:      npm-private-packages: 501ms
client-release-group-root:      npm-package-license: 474ms
client-release-group-root:      npm-package-json-prettier: 464ms
client-release-group-root:      npm-package-json-script-clean: 457ms
client-release-group-root:      npm-package-json-script-dep: 437ms
client-release-group-root:      npm-package-json-scripts-args: 436ms
client-release-group-root:      npm-package-json-test-scripts: 395ms
client-release-group-root:      npm-package-json-script-mocha-config: 346ms
client-release-group-root:      npm-package-json-script-jest-config: 386ms
client-release-group-root:      npm-package-json-esm: 348ms
client-release-group-root:      npm-package-json-clean-script: 342ms
client-release-group-root:      npm-package-types-field: 331ms
client-release-group-root:      npm-package-exports-field: 136ms
client-release-group-root:      fluid-build-tasks-tsc: 240ms
client-release-group-root:      fluid-build-tasks-eslint: 156ms
client-release-group-root:      pnpm-npm-package-json-preinstall: 1ms
client-release-group-root:      pnpm-lock-workspace: 0ms
client-release-group-root:      pnpm-lock-no-package-lock: 0ms
client-release-group-root:      assert-short-codes: 2633ms
client-release-group-root:      html-copyright-file-header: 2ms
client-release-group-root:      dockerfile-packages: 18ms
client-release-group-root:      package-lockfiles-no-private-url: 1ms
client-release-group-root: Performance for "final":
client-release-group-root:      assert-short-codes: 17ms
[13/13] x client-release-group-root: flub check policy - 5.104s
~~~

Reply:
Ok, I think instead of adding a `"build:test"` script that does the exact same thing as the `"tsc"` one, we can do [what we did in the common-definitions package](https://github.com/microsoft/FluidFramework/blob/25406b773364cd47e0d57d31db16bc5e94a9cd7d/common/lib/common-definitions/package.json#L75-L80), to explicitly declare that the `tsc` task depends on the `typetests:gen` one. The problem that the policy is complaining about is that if `build:test` doesn't exist but `typetests:gen` does, then we need to make sure that the build of the source code depends on generating the typests, i.e. `typetests:gen` (when `build:test` exists, [our policy already ensures _that one_ depends on `typetests:gen`](https://github.com/microsoft/FluidFramework/blob/25406b773364cd47e0d57d31db16bc5e94a9cd7d/fluidBuild.config.cjs#L54-L58)).

Reply:
I see! That's what it means!! Updating...Thx, Alejandro